dkg-gadget/src/worker.rs

```diff
@@ -1001,21 +1007,35 @@ where
 								reason: err.to_string(),
 							})
 						}
+						return Ok(())
 					}
 				}
+
+				Err(DKGError::GenericError {
+					reason: "Message is not for this DKG round or DKG rounds are not ready yet"
+						.into(),
+				})
 			},
 			DKGMsgPayload::Offline(..) | DKGMsgPayload::Vote(..) => {
 				let msg = Arc::new(dkg_msg);
 				let async_index = msg.msg.payload.get_async_index();
+				log::debug!(target: "dkg_gadget::worker", "Received message for async index {}", async_index);
 				if let Some(rounds) = self.signing_rounds[async_index as usize].as_mut() {
```

Comment:
Let's nip this overflow error here in this PR since it should be quick to add.

Reply:
We will re-architect this part and hopefully split it up to different parts in a different pr